test/net/sourceforge/kolmafia/request/RelayRequestTest.java

```diff
@@ -1,13 +1,32 @@
 package net.sourceforge.kolmafia.request;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.io.File;
+import net.sourceforge.kolmafia.KoLConstants;
 import net.sourceforge.kolmafia.preferences.Preferences;
+import net.sourceforge.kolmafia.utilities.FileUtilities;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 public class RelayRequestTest {
 
+  /**
+   * The global clean up code will delete root/relay since relay is not under Git control. This is
+   * the only test that breaks when that happens. Duplicate the code that loads the relay directory
+   * here so that the directory will be present when the test is run.
+   */
+  @BeforeEach
+  public void initializeRelayFileDirectory() {
```

Comment:
this is only need for `findVariousRelayFilesOrNot` so can be moved to inside that method.

Reply:
Philosophy.  Do I put it in BeforeEach in anticipation of a time when there are other tests that will almost certainly need it or do I inline it and hope that the opportunity to extract and reuse will be noticed?  I'll make the change but BeforeEach was a choice, perhaps not the best one :-)